addon/mixins/model-records/relationship-route-mixin.js

```diff
@@ -1,26 +1,34 @@
 import Ember from 'ember'
 
 const {
-  Mixin,
-  String: { singularize }
+  Mixin
 } = Ember
 
 export default Mixin.create({
   setupController(controller, model) {
     this._super(controller, model)
 
     controller.setProperties({
-      parentModel: this.modelFor('admin.model-records.show'),
-      recordName: this._relationshipName(),
+      parentModel: this._parentModel(),
+      relationshipName: this._relationshipName(),
       recordType: this._relationshipType()
     })
   },
 
+  _parentModel() {
+    return this.modelFor('admin.model-records.show')
+  },
+
   _relationshipName() {
     return this.paramsFor('admin.model-records.show.related').relationship_name
   },
 
   _relationshipType() {
-    return singularize(this._relationshipName())
+    const store = this.get('admin.store')
```

Comment:
this is also assuming admin is defined - what is admin here?

Reply:
looks like its a service and it should be set directly on here since it is required